docs/system/k8s-metrics.md

```diff
@@ -329,5 +377,347 @@ This metric is [recommended][MetricRecommended].
 <!-- END AUTOGENERATED TEXT -->
 <!-- endsemconv -->
 
+## Deployment Metrics
+
+**Description:** Deployment level metrics captured under the namespace `k8s.deployment`.
+
+### Metric: `k8s.deployment.desired_pods`
+
+This metric is [recommended][MetricRecommended].
+
+<!-- semconv metric.k8s.deployment.desired_pods -->
+<!-- NOTE: THIS TEXT IS AUTOGENERATED. DO NOT EDIT BY HAND. -->
+<!-- see templates/registry/markdown/snippet.md.j2 -->
+<!-- prettier-ignore-start -->
+<!-- markdownlint-capture -->
+<!-- markdownlint-disable -->
+
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `k8s.deployment.desired_pods` | Gauge | `{pod}` | Number of desired replica pods in this deployment [1] | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```

Comment:
Should we explicitly clarify that the `this deployment` refers to the specific deployment resource this metric is for?

Reply:
I see there is a note earlier in this doc which states to refer to the relevant k8s resource.

Reply:
Yeap, probably we will need to iterate on this to improve the note: https://github.com/open-telemetry/semantic-conventions/pull/1649#discussion_r1872258043